r1cs/gadgets/crypto/src/merkle_tree/mod.rs

```diff
@@ -119,7 +119,10 @@ where
         leaf: impl ToBytesGadget<ConstraintF>,
         should_enforce: &Boolean,
     ) -> Result<(), SynthesisError> {
-        debug_assert!(self.path.len() == P::HEIGHT);
+        // TODO: check error message
+        if self.path.len() != P::HEIGHT {
+            return Err(SynthesisError::Other("path length verification failed".to_owned()));
```

Comment:
"Path length must be equal to height. Path len: {}, Height: {}", path.len(), P::HEIGHT